Source/quests.cpp

```diff
@@ -189,10 +162,7 @@ void DrawLTBanner(Point position)
 	int width = SDL_SwapLE16(dunData[0]);
 	int height = SDL_SwapLE16(dunData[1]);
 
-	setpc_x = position.x;
-	setpc_y = position.y;
-	setpc_w = width;
-	setpc_h = height;
+	SetPiece = { position, { SDL_SwapLE16(dunData[0]), SDL_SwapLE16(dunData[1]) } };
 
 	const uint16_t *tileLayer = &dunData[2];

```

Comment:
I see this one is unique in that it doesn't protect the region, so later level generation steps can trample it.

Reply:
This actually happens after the level has been fully generated. The position for it is decided during `FillChambers()` where `Banner1.dun` is placed (and then copied to `pdungeon[]`) before being overwritten with `Banner2.dun` at this stage, which is the one used for the initial level state before you complete the quest. Because it is always placed inside one of the 3 chambers it would already have been protected from the wall generators.

Functions that could potentially overwrite the `pdungeon[]` version:
`FixCornerTiles()`
`ApplyShadowsPatterns()`
`FillFloor()`

Reply:
Oooooh, I see now. `pdungeon != dungeon` 😂 . No wonder that doesn't use `PlaceDunTiles`

Reply:
yeah, this took me a while to figure out how i broke things. For some reason that one quest is loaded in reverse order of literally all other quests where `pdungeon` is handled in a general way.